tools/state-viewer/src/replay_headers.rs

```diff
@@ -67,23 +73,143 @@ pub(crate) fn replay_headers(
+    for account_id in left_accounts.union(&right_accounts) {
+        let left_summary = left_summaries.get(account_id);
+        let right_summary = right_summaries.get(account_id);
+        if left_summary.is_some() ^ right_summary.is_some() {
+            println!("{}: {:?} --> {:?}", account_id.as_str(), left_summary, right_summary);
+            continue;
+        }
+        let left_summary = left_summary.unwrap();
+        let right_summary = right_summary.unwrap();
+        if left_summary.stake != right_summary.stake {
+            println!(
+                "{}: Stake [diff={}] {} --> {}",
+                account_id.as_str(),
+                right_summary.stake - left_summary.stake,
+                left_summary.stake,
+                right_summary.stake
+            );
+        }
+        if left_summary.block_production_rate != right_summary.block_production_rate {
+            println!(
+                "{}: Blocks [diff=%{}] %{} --> %{}",
+                account_id.as_str(),
+                right_summary.block_production_rate - left_summary.block_production_rate,
+                left_summary.block_production_rate,
+                right_summary.block_production_rate
+            );
+        }
+        if left_summary.chunk_production_rate != right_summary.chunk_production_rate {
+            println!(
+                "{}: Chunks [diff=%{}] %{} --> %{}",
+                account_id.as_str(),
+                right_summary.chunk_production_rate - left_summary.chunk_production_rate,
+                left_summary.chunk_production_rate,
+                right_summary.chunk_production_rate
+            );
+        }
+        if left_summary.chunk_endorsement_rate != right_summary.chunk_endorsement_rate {
+            println!(
+                "{}: Endorsements [diff=%{}] %{} --> %{}",
+                account_id.as_str(),
+                right_summary.chunk_endorsement_rate - left_summary.chunk_endorsement_rate,
+                left_summary.chunk_endorsement_rate,
+                right_summary.chunk_endorsement_rate
+            );
+        }
+    }
+}
+
+fn compare_validator_kickout_stats(left: &EpochValidatorInfo, right: &EpochValidatorInfo) {
+    let left_kickouts = get_validator_kickouts(left);
+    let right_kickouts = get_validator_kickouts(right);
+    let left_accounts: HashSet<AccountId> = left_kickouts.keys().cloned().collect();
+    let right_accounts: HashSet<AccountId> = right_kickouts.keys().cloned().collect();
+    for account_id in left_accounts.union(&right_accounts) {
+        let left_kickout = left_kickouts.get(account_id);
+        let right_kickout = right_kickouts.get(account_id);
+        if left_kickout.is_some() ^ right_kickout.is_some() {
+            println!("{}: {:?} --> {:?}", account_id.as_str(), left_kickout, right_kickout);
+            continue;
+        }
+        let left_kickout = left_kickout.unwrap();
+        let right_kickout = right_kickout.unwrap();
```

Comment:
Can those both be None here?

Reply:
Both cannot be None because the loop is over the union of the keys of the hashmaps, so the key should be in one of the hashmaps.